### PR TITLE
Migrated NodePalette_HappyPath_ClearSelection to QtPy

### PR DESCRIPTION
**Pull request by michleza**
Signed-off-by: Michael <michleza@amazon.com>

## What does this PR do?
updated test script to use QtPy framework
updated QtPy NodePalette class to include a clear button function 
added a constant to the scripting constants file

## How was this PR tested?
pytest via CLI
python console via O3DE editor

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/scripting/NodePalette_HappyPath_ClearSelection.py [hunk 1]
```diff
@@ -23,56 +21,45 @@ def NodePalette_HappyPath_ClearSelection():
...
-    Report.result(Tests.search_string_cleared, search_box.text() == "")
+
+    # 1) Initialize our O3DE editor object
+    qtpy_o3de_editor = QtPyO3DEEditor()
+
+    # 2) Open and initialize Script Canvas window (Tools > Script Canvas)
+    qtpy_o3de_editor.open_script_canvas()
+
+    # 3) Get a local handle on the Node Palette QtPy object
+    qtpy_node_palette = qtpy_o3de_editor.sc_editor.node_palette
```
Comment (FuzzyCarterAWS) [inline AutomatedTesting/Gem/PythonTests/scripting/NodePalette_HappyPath_ClearSelection.py:53]: Move this up to step 2 and reword it a bit along the lines of "Open Script Canvas Window and initialize objects for test.

don't forget to renumber the steps here and in the desc.

**Review by FuzzyCarterAWS — CHANGES_REQUESTED** [review]

File: AutomatedTesting/Gem/PythonTests/scripting/NodePalette_HappyPath_ClearSelection.py (hunk 1 above)
Comment (michleza) [reply AutomatedTesting/Gem/PythonTests/scripting/NodePalette_HappyPath_ClearSelection.py:53]: moved this up and renumber/reworded the steps

**Review by FuzzyCarterAWS — APPROVED** [review]
LGTM

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasNodePalette.py [hunk 2]
```diff
@@ -9,8 +9,7 @@
 from editor_python_test_tools.utils import TestHelper as helper
 from PySide2 import QtWidgets, QtTest, QtCore
 import pyside_utils
-import azlmbr.legacy.general as general
-from consts.scripting import (NODE_PALETTE_QT, TREE_VIEW_QT, SEARCH_FRAME_QT, SEARCH_FILTER_QT)
+from consts.scripting import (NODE_PALETTE_QT, TREE_VIEW_QT, SEARCH_FRAME_QT, SEARCH_FILTER_QT, NODE_PALETTE_CLEAR_BUTTON_QT)
 from consts.general import (WAIT_TIME_SEC_1, WAIT_TIME_SEC_3)
 
 class QtPyScriptCanvasNodePalette():
```
Comment (cgalvan) [inline AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasNodePalette.py:15]: Not necessarily in scope of this PR, but this node palette is actually in the base `GraphCanvas` gem and is used by our other *Canvas tools as well (e.g. Landscape Canvas, Material Canvas, etc...) so might be worth renaming this to `QtPyGraphCanvasNodePalette`

**Review by cgalvan — APPROVED** [review]

**Pull request merged**